cli/tests/test_resolve_command.rs

```diff
@@ -949,6 +949,159 @@ fn test_description_with_dir_and_deletion() {
     "###);
 }
 
+#[test]
+fn test_resolve_conflicts_with_executable() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    // Create a conflict in "file1" where all 3 terms are executables, and create a
+    // conflict in "file2" where one side set the executable bit.
+    create_commit(
+        &test_env,
+        &repo_path,
+        "base",
+        &[],
+        &[("file1", "base1\n"), ("file2", "base2\n")],
+    );
+    test_env.jj_cmd_ok(&repo_path, &["file", "chmod", "x", "file1"]);
+    create_commit(
+        &test_env,
+        &repo_path,
+        "a",
+        &["base"],
+        &[("file1", "a1\n"), ("file2", "a2\n")],
+    );
+    create_commit(
+        &test_env,
+        &repo_path,
+        "b",
+        &["base"],
+        &[("file1", "b1\n"), ("file2", "b2\n")],
+    );
+    test_env.jj_cmd_ok(&repo_path, &["file", "chmod", "x", "file2"]);
+    create_commit(&test_env, &repo_path, "conflict", &["a", "b"], &[]);
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["resolve", "--list"]), 
+    @r#"
+    file1    2-sided conflict including an executable
+    file2    2-sided conflict including an executable
+    "#);
+    insta::assert_snapshot!(
+        std::fs::read_to_string(repo_path.join("file1")).unwrap(),
+        @r##"
+    <<<<<<< Conflict 1 of 1
+    %%%%%%% Changes from base to side #1
+    -base1
+    +a1
+    +++++++ Contents of side #2
+    b1
+    >>>>>>> Conflict 1 of 1 ends
+    "##
+    );
+    insta::assert_snapshot!(
+        std::fs::read_to_string(repo_path.join("file2")).unwrap(),
+        @r##"
+    <<<<<<< Conflict 1 of 1
+    %%%%%%% Changes from base to side #1
+    -base2
+    +a2
+    +++++++ Contents of side #2
+    b2
+    >>>>>>> Conflict 1 of 1 ends
+    "##
+    );
+    let editor_script = test_env.set_up_fake_editor();
+
+    // Test resolving the conflict in "file1", which should produce an executable,
+    // but currently doesn't
```

Comment:
nit: Rewrite it as a TODO comment so its immediately clear that it is fixed by the next commit.